zerver/views/message_edit.py

```diff
@@ -128,6 +128,12 @@ def update_message_backend(request: HttpRequest, user_profile: UserMessage,
     else:
         raise JsonableError(_("You don't have permission to edit this message"))
 
+    # Disallow editing of messages in deactivated streams unless you are an admin.
+    if message.is_stream_message() and not user_profile.is_realm_admin:
+        message_stream = get_stream_by_id(message.recipient.type_id)
+        if message_stream.deactivated:
+            return json_error(_("This message is in a deactivated stream"))
```

Comment:
A better message could be "Messages in deactivated streams cannot be edited." (likewise for deletion below)

Reply:
I agree, that is definitely more informative. I will make that change shortly.